### DISCUSSION TITLE
Access location in links function

### DISCUSSION THREAD

**Discussion by liegeandlief** [remix-run/remix#2276] [Q&A] [answered]
Is it possible to access the `location` object in the `links` function? It is possible in the `meta` function but I can't seem to access it in the `links` function.

I need the `location` object in order to generate canonical and hreflang link tags.

**Comment by sergiodxa** [comment, marked as answer]
LinksFunction doesn't receive any param, this is required by Remix so it can prefetch those links.

Most links depending on data are body safe so they can be used inside the body. In case it's not possible, you can use https://github.com/sergiodxa/remix-utils#dynamiclinks to have links based on data, then return the location from the loader.

**Comment by liegeandlief** [comment]
Unfortunately the `canonical` and `alternate` link types are not `body-ok` (https://html.spec.whatwg.org/multipage/links.html#body-ok).

Looks like I'll need to use that dynamic links utility.
